### ISSUE TITLE
Bug/False-Positive: Trim incorrectly suggested for regex 

### ISSUE THREAD

**Issue by dstockto** [kalessil/phpinspectionsea#1145]
First, thank you for reporting a bug or making a request. That takes time and we appreciate that!

Let's start with common details:
 
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug/False-Positive                                       |
| Plugin         | Php Inspections (EA Extended)  |
| Language level | PHP 7.3                        |

Current behaviour (description/screenshot:)
--

With this code snippet:

```
return preg_replace('/^\s+|\s+$/u', '', $request);
```

The plugin suggests that it can be replaced with a call to trim, but trim will not result in the same behavior for unicode whitespace: "\n\r \t\u{000b}\u{00a0}\u{2002}"

Expected behaviour
--

If the unicode /u flag on the regex is found, it should not suggest trim since trim doesn't account for unicode it doesn't work the same way.

Environment details
---
Information from Help -> About dialog (contains IDE and its version details).

PhpStorm 2018.3
Build #PS-183.4284.150, built on November 21, 2018

You have a perpetual fallback license for this version
Subscription is active until December 9, 2019
JRE: 1.8.0_152-release-1343-b15 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.13.6

Notes
---
- Bugs fixed with the highest priority, so features might be postponed. But it shouldn't stop you, right?
- The project support is powered by community funding via [Patreon](https://www.patreon.com/kalessil)

**Comment by kalessil** [comment]
Thank you for reporting @dstockto, fixed!
